pimd/pim_cmd.c

```diff
@@ -3007,12 +3011,18 @@ static void pim_show_bsm_db(struct pim_instance *pim, struct vty *vty, bool uj)
 
 		if (uj) {
 			json_object_string_add(json, "BSR address", bsr_str);
+			json_object_string_add(json, "bsrAddress", bsr_str);
```

Comment:
Are you planning to print both BSR address and bsrAddress in the json o/p

Reply:
At the moment, yes and it basically breaks automated stuff if you are using `<string, dict>` iteration over this object. We could leave only a single, but it's bad also... How do you suggest continuing?

Reply:
Yes, it looks bad, but I am not sure on this.
Because for the people who is checking "BSR address", it will fail for them if we remove this

Reply:
How does it break iteration? Adding new keys should be safe, we do this all the time.

Reply:
It wont break things, But it will look weird if we print 2 times.
But to pass things this is the only option we have.

Reply:
@idryzhov you can check here https://github.com/FRRouting/frr/pull/10447/commits/dbe8d09d8d11008e87bf89ed54ec56d63fa582a4

It expects a tuple of `<string, dict>`, but finds `<string, string, dict>`.

Reply:
I see. The script is just badly written, I don't think we should cover all bad scripts in the world :)

BTW, the main problem here is that this whole JSON output is completely incorrect. It uses IP address as a key and this is wrong, we must never use changeable objects as keys.

Instead of
```
{
  "bsrAddress": "1.1.2.7",
  "225.200.100.100/32": {
    "210.210.210.210": {
      "rpAddress": "210.210.210.210",
      "rpHoldTime": 150,
      "rpPriority": 0,
      "hashVal": 833644616
    },
    "pendingRpCount": 0
  }
}
```
It should be something like
```
{
  "bsrAddress": "1.1.2.7",
  "groups": [
    {
      "group": "225.200.100.100/32",
      "rps": [
        {
          "rpAddress": "210.210.210.210",
          "rpHoldTime": 150,
          "rpPriority": 0,
          "hashVal": 833644616
        }
      ],
      "pendingRpCount": 0
    }
  ]
}
```

Reply:
Yep.

Reply:
Hi @ton31337 ,
We should keep only camelCase values, it will avoid rework in future.
For breaking scripts consideration, you have already modified script logic to use camelCase values. I think that will work fine.